homeassistant/helpers/llm.py

```diff
@@ -354,28 +354,40 @@ def _async_get_preable(self, llm_context: LLMContext) -> list[str]:
         ]
         area: ar.AreaEntry | None = None
         floor: fr.FloorEntry | None = None
+        device_name_prompt = None
+
         if llm_context.device_id:
             device_reg = dr.async_get(self.hass)
             device = device_reg.async_get(llm_context.device_id)
 
             if device:
                 device_name = device.name_by_user or device.name
-                if device_name:
-                    prompt.append(f"User is interacting with you from a device called {device_name}.")
+                device_name_prompt = f"User is interacting with you via {device_name}"
```

Comment:
Does everything still work if we keep this in the existing format?

"You are in area {area}"
"You are device {device} in area {area} (foor {floor})"

Reply:
No, gpt4o-mini totally clueless, claude3.5 and gpt 4o sometimes realize this means the device name (like if the device name is "Voice Assistant PE in Office", but are clueless if the name is more ambiguous "Satellite Office"). Saying explicitly that "User is interacting with you via XXX" solves the issue on 4o, 4o-mini and claude3.5 with any device name I tried.

Reply:
OK, my primary motification was to make this easier to follow. I have another suggestion i'll make instead then.